src/TSR.cpp

```diff
@@ -107,13 +106,49 @@ bool TSR::deserialize(std::stringstream &ss) {
     return _initialized;
 }
 
+void TSR::serialize(std::ostream& ss)
+{
+    static const int manipind_ignored = 0;
+    static const std::string relativebodyname_ignored = "NULL";
+    static const std::string relativelinkname_ignored = "";
+
+    if (!_initialized)
+        throw std::runtime_error("TSR is not initialized.");
+
+    ss << manipind_ignored
+       << ' ' << relativebodyname_ignored;
+
+    if( relativebodyname_ignored != "NULL")
+        ss << ' ' << relativelinkname_ignored;
```

Comment:
I now store those values in member variables. This should make `deserialize(serialize(x))` correctly propagate the values. Even though we don't use them anywhere. 😃